File: C++/数学/Sieve_E.cpp
constexpr int M = 2e8;
bool is_prime[M];
vector<int> primes;
auto init = [] {
    ranges::fill(is_prime, true);
    is_prime[0] = is_prime[1] = false;
    for (int i = 2; i < M; i++) {
        if (is_prime[i]) {
            primes.push_back(i);
            for (ll j = 1LL * i * i; j < M; j += i) {
                is_prime[j] = false;
            }
        }
    }
    return 0;
}();
// 函数指针在创建时自动调用

File: Python/数学/Sieve_E.py
M = 2 * 10**8
is_prime = bytearray()
primes = []


def init() -> None:
    global is_prime

    is_prime = bytearray(b"\x01") * M
    is_prime[0] = is_prime[1] = False
    primes.clear()
    for i in range(2, M):
        if is_prime[i]:
            primes.append(i)
            for j in range(i * i, M, i):
                is_prime[j] = False


# Python 中避免导入文件时立即分配约 200 MB，使用前手动调用 init()
# 用法：init()
